lib/ansible/module_utils/rds.py

```diff
@@ -0,0 +1,102 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+import botocore
+
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+
+
+def get_db_instance(conn, instancename):
+    try:
+        response = conn.describe_db_instances(DBInstanceIdentifier=instancename)
+        instance = RDSDBInstance(response['DBInstances'][0])
+        return instance
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBInstanceNotFound':
+            return None
+        else:
+            raise
+
+
+def get_db_snapshot(conn, snapshotid):
+    try:
+        response = conn.describe_db_snapshots(DBSnapshotIdentifier=snapshotid)
+        snapshot = RDSSnapshot(response['DBSnapshots'][0])
+        return snapshot
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBSnapshotNotFound':
+            return None
+        else:
+            raise
+
+
+class RDSDBInstance(object):
+    def __init__(self, dbinstance):
+        self.instance = dbinstance
+        self.name = self.instance.get('DBInstanceIdentifier')
+        self.status = self.instance.get('DBInstanceStatus')
+        self.data = self.get_data()
+
+    def get_data(self):
+        return camel_dict_to_snake_dict(self.instance)
+
+    def diff(self, params):
+        compare_keys = ['backup_retention', 'instance_type', 'iops',
+                        'maintenance_window', 'multi_zone',
+                        'replication_source',
+                        'size', 'storage_type', 'tags', 'zone']
+        before = dict()
+        after = dict()
+        for k in compare_keys:
+            if self.params.get(k) != params.get(k):
+                before[k] = self.params.get(k)
+                after[k] = params.get(k)
+        port = self.data.get('port')
+        if port != params.get('port'):
+            before['port'] = port
+            after['port'] = params.get('port')
+        result = dict()
+        if before:
+            result = dict(before_header=self.name, before=before, after=after)
+            result['after_header'] = params.get('new_instance_name', params.get('instance_name'))
+        return result
+
+    def __eq__(self, other):
+        return not self.diff(other.instance)
+
+    def __ne__(self, other):
+        return not self == other
+
+
+class RDSSnapshot(object):
```

Comment:
Same - this class seems more like a snapshot -> facts dict function than its own class.